### PR TITLE
Add "renderer" environment variable #33463

### PR DESCRIPTION
**Pull request by lucypero**
Added "renderer" environment variable so the user and platform can conditionally compile each renderer's code, or none at all.

#33463

### CODE REVIEW

File: SConstruct [hunk 1]
```diff
@@ -120,6 +120,7 @@ opts.Add(BoolVariable('deprecated', "Enable deprecated features", True))
 opts.Add(BoolVariable('gdscript', "Enable GDScript support", True))
 opts.Add(BoolVariable('minizip', "Enable ZIP archive support using minizip", True))
 opts.Add(BoolVariable('xaudio2', "Enable the XAudio2 audio driver", False))
+opts.Add(EnumVariable('renderer', "Select Renderer", 'vulkan', ('vulkan', 'gles2', 'gles3', 'none')))
```
Comment (aaronfranke) [inline SConstruct]: GLES 3 is going to be removed immediately when Vulkan is added, it's best to remove it and then this PR will just wait until the Vulkan branch is merged and GLES 3 is removed.

Comment (MatiasPi) [reply SConstruct]: Done

**Comment by akien-mga** [comment]
This is not useful as is, as the main use case is to compile both rendering backends and this breaks it.

It's likely also too early to implement this as the gles2 backend is not working in the `vulkan` branch.

**Comment by akien-mga** [comment]
Closing as per above comment.

**Pull request closed without merging**